src/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -81,14 +81,7 @@ protected override void SetAdditionalRequestParameters(OAuth2Client client)
 
                 RequestContext.Logger.Verbose("Refreshing access token...");
                 await base.SendTokenRequestAsync().ConfigureAwait(false);
-
-                if (Response.IdToken == null)
```

Comment:
Wooohoo! does it mean that we have an agreement with service team that id token will always be returned?